src/os/bluestore/BlueStore.cc

```diff
@@ -6322,11 +6323,19 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 		   << dendl;
 	} else {
 	  _txc_finalize_kv(txc, txc->t);
+	  if (g_conf->bluestore_rtc && !txc->is_pipelined_io && txc->wal_txn) {
+	    if (txc->osr->kv_finisher_submitting == 1) {
```

Comment:
Without txc->wal_txn,  ShardedOpWQ holding osr->qlock runs to _txc_finish and _osr_reap_done method which also requires osr->qlock.

I think i can solve the problem by not trying to take osr->qlock if there is only metadata operation..
I'll change the code.

Thank you for your opinion